### PR TITLE
Thena-like gauges

### PR DESCRIPTION
**Pull request by vminkov**

### CODE REVIEW

File: contracts/Gauge.sol [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+    IERC20 public rewardToken;
+
+    // used in the inheriting contracts
+    address public target;
+
+    // TODO unused?
+    address public ve;
+    address public distribution;
+    address public gaugeRewarder;
```
Comment (vminkov) [inline contracts/Gauge.sol:38]: @qzirak : is the `gaugeRewarder` used?

File: contracts/Gauge.sol [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    }
+
+    function initialize(
+        address _rewardToken,
+        address _ve,
+        address _target,
+        address _distribution,
+        address _internal_bribe,
+        address _external_bribe
+    ) external initializer {
```
Comment (qzirak) [inline contracts/Gauge.sol:95]: initializer should be changed to onlyInitializing

Comment (vminkov) [reply contracts/Gauge.sol:95]: `onlyInitializing` is applicable for only for the sub-calls of the root `initialize` call e.g. `__Ownable_init()`

the inheriting `MarketGauge` and `PairGauge` don't need to have their own `initialize` fn that calls the `Gauge:initialize` fn

File: contracts/Voter.sol [hunk 3]
```diff
@@ -529,74 +516,55 @@ contract VoterV3 is IVoter, OwnableUpgradeable, ReentrancyGuardUpgradeable {
...
 
     /// @notice create a gauge
-    /// @param  _pool       LP address 
+    /// @param  _target  gauge target address
     /// @param  _gaugeType  the type of the gauge you want to create
     /// @dev    To create stable/Volatile pair gaugeType = 0, Concentrated liqudity = 1, ...
     ///         Make sure to use the corrcet gaugeType or it will fail
 
-    function _createGauge(address _pool, uint256 _gaugeType) internal returns (address _gauge, address _internal_bribe, address _external_bribe) {
+    function _createGauge(address _target, uint256 _gaugeType) internal returns (address _gauge, address _internal_bribe, address _external_bribe) {
```
Comment (qzirak) [inline contracts/Voter.sol:524]: What the _gaugeType represents in the PR version ?

File: contracts/Gauge.sol (hunk 1 above)
Comment (qzirak) [reply contracts/Gauge.sol:38]: For thena, I have only seen 0 addresses for their gauges

File: contracts/Voter.sol (hunk 3 above)
Comment (vminkov) [reply contracts/Voter.sol:524]: the `_gaugeType` is removed in the other PRs

File: contracts/Gauge.sol (hunk 1 above)
Comment (vminkov) [reply contracts/Gauge.sol:38]: let's agree to remove this var then?

Comment (qzirak) [reply contracts/Gauge.sol:38]: I think let's remove this rewarder

**Pull request merged**